src/components/Select/types.ts

```diff
@@ -0,0 +1,29 @@
+export type ISelectChange = (value: { country: string; code: string }) => void;
+
+export interface ISelectItem {
+  label: string;
+  value: string;
+}
+
+export interface ISelectProps {
+  clickedOutside?: boolean;
+  defaultValue?: ISelectItem;
+  label?: string;
+  onChange?: (event: any) => void;
+  options: ISelectItem[];
+  name: string;
+  autoComplete?: string;
+  setElementRef?: (
+    el: React.RefObject<HTMLElement>
+  ) => React.RefObject<HTMLDivElement>;
+}
+
+export interface IListArgs {
+  options: ISelectItem[];
+  onChange: ISelectChange;
+}
+
+export interface IFilteredListArgs {
+  options: ISelectItem[];
+  searchPhrase: string;
+}
```

Comment:
I think name of this file might be confusing for some. Apollo codegen is configured to creates types in `types/` folder. 
Consider changing one or other.